master/usage/assigning-ip-addresses-topology.md

```diff
@@ -0,0 +1,159 @@
+---
+title: Assigning IP addresses based on topology
+---
+
+## Why this is useful
+
+{{site.prodname}} can be configured to use specific IP pools for different
+topological areas. For example, you may want workloads in a particular rack,
+zone, or region to receive addresses from the same IP pool. This may be
+desirable either to reduce the number of routes required in the network or to
+meet requirements imposed by an external firewall device or policy. This guide
+walks through an example of how to set this up.
+
+## How this works
+At a high level, this feature is operated by the setting node labels and then
+selecting those nodes via node selector on your IP pool resources. Note that
+there are a variety of other ways to specify IP address assignment behavior that
+can be found in the [cni-plugin configuration reference document]({{site.baseurl}}/{{page.version}}/reference/cni-plugin/configuration).
+Each approach has its own use-case but this particular one takes on the lowest
+priority compared to the others. Specifying IP pools through
+configuration takes precedence over IP addresses through annotations, which
+takes precedence over IP pool assignment through node selectors.
+
```

Comment:
Is this a good point to mention that if there are no available IPs in a pool for a node, it will be unable to setup pods?

Reply:
I've made a note to reflect that point at the bottom of this page. Let me know what you think https://github.com/projectcalico/calico/pull/2371/files#diff-a13aac29e4b668cc996f537412039845R150